src/common/neon_sse.h

```diff
@@ -0,0 +1,84 @@
+#ifndef NEON_SSE_H
+#define NEON_SSE_H
+
+#include <arm_neon.h>
+#include <stdlib.h>
+
+typedef int32x4_t	__m128i;
+typedef float32x4_t	__m128;
+
+static __inline void *
+__mm_malloc(size_t size, size_t alignment)
+{
+	void *ptr;
+
+	if (posix_memalign(&ptr, alignment, size) == 0)
+		return ptr;
+	else
+		return NULL;
+}
+
+static __inline void
+__mm_free(void *ptr)
+{
+	free(ptr);
+}
+
+static __inline void
+_mm_clflush(void const *p __attribute__((unused)))
```

Comment:
I'm not sure an empty clflush and clwb are a good choice. At least crash with a sensible error?

Reply:
I agree, when I see the `_mm` intrinsics I assume they are x86 intrinsics. This looks very misleading, we should rather have a wrapper header with something like:
```
static inline flush_cache_line(const void *p)
{
  abort(); // or something similar
}
```

It would also be nice to have an AARCH64 variant of `is_cpu_clflushopt_present`, which returns zero (why isn't that bool BTW?? ) -- in a conditionally compiled file.
I.e. splitting `src/libpmem/cpu.c` to `src/libpmem/cpu_aarch64.c` and `src/libpmem/cpu_x86_64.c`, and let make choose which one to include in the build.

Reply:
I am not sure about crashing, unfortunately __builtin__clear_cache cannot be used here. Even with cacheflush routine, we would get the discards ‘const’ qualifier  error. Any alternatives with compiler intrinsics that can support this?

Reply:
There should be no problem with crashing.
If a `is_cpu_clflushopt_present` function indicates that it is not available, it will never be called.
An empty (or rather crashing) wrapper function would just allow us to have fewer ifdef's in `src/libpmem/pmem.c` -- also, test would notice pretty soon if it is called by accident.
An empty function has the only disadvantage, that a test doesn't notice such a problem so easily.

If this empty function is called by mistake, it would be more difficult to notice.